archiver.go

```diff
@@ -19,6 +19,7 @@ type Archiver interface {
 	// are walked and recursively added, preserving folder
 	// structure.
 	Archive(sources []string, destination string) error
+	CheckExt(name string) error
```

Comment:
This might sound pedantic, but how about a new interface called `ExtChecker` or `ExtensionChecker` with this method in it, then just embed that interface into Archiver and Compressor.

Reply:
Gotcha, makes sense!